test/unit/remote-config/remote-config-api-client.spec.ts

```diff
@@ -388,13 +304,44 @@ describe('RemoteConfigApiClient', () => {
       });
     });
 
+    VALIDATION_ERROR_MESSAGES.forEach((message) => {
+      it('should reject with failed-precondition when a validation error occurres', () => {
+        const stub = sinon
+          .stub(HttpClient.prototype, 'send')
+          .rejects(utils.errorFrom({
+            error: {
+              code: 400,
+              message: message,
+              status: "FAILED_PRECONDITION"
+            }
+          }, 400));
+        stubs.push(stub);
+        const expected = new FirebaseRemoteConfigError('failed-precondition', message);
+        return apiClient.publishTemplate(REMOTE_CONFIG_TEMPLATE)
+          .should.eventually.be.rejected.and.deep.equal(expected);
+      });
+    });
+  });
+
+  function runApiResponseValidationTests(rcOperation: () => Promise<RemoteConfigTemplate>): void {
```

Comment:
Most of this is checking for various error responses. So may be `runErrorResponseTests()` is more appropriate? Your call.